parcels/particleset/particlesetsoa.py

```diff
@@ -470,17 +553,20 @@ def add(self, particles):
         if isinstance(particles, BaseParticleSet):
             particles = particles.collection
         self._collection += particles
+        self._dirty_neighbor = True
         return self
 
     def remove_indices(self, indices):
         """Method to remove particles from the ParticleSet, based on their `indices`"""
+        self._dirty_neighbor = True
```

Comment:
What happens if this is set to False? Is this still debugging comment?

Reply:
It is not a debugging comment. It ensures that the next time, the neighbor tree is rebuilt.